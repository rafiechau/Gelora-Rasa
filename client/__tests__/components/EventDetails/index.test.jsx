import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import EventDetail from '@components/EventDetails';

jest.mock('react-intl', () => ({
  FormattedMessage: jest.fn(({ id }) => <span>{id}</span>),
}));

describe('EventDetail', () => {
  const mockEvent = {
    id: 1,
    eventName: 'Musical Night',
    date: '2024-01-01',
    time: '20:00',
    price: 50000,
    address: 'Jalan Raya No.123, Jakarta',
    type: 'hybrid', // 'offline', 'online', atau 'hybrid'
    Location: {
      namaProvinsi: 'Jakarta',
    },
  };
  const formattedDate = '01 Jan 2024';
  const handleTicketQuantityChange = jest.fn();
  const ticketQuantity = 1;
  const selectedTicketType = 'offline';
  const handleTicketTypeChange = jest.fn();
  const handleOrder = jest.fn();
  const countdown = '5 days';
  const canOrder = true;
  const hasOrdered = false;
  const user = { role: 1 };

  beforeEach(() => {
    render(
      <EventDetail
        event={mockEvent}
        formattedDate={formattedDate}
        handleTicketQuantityChange={handleTicketQuantityChange}
        ticketQuantity={ticketQuantity}
        selectedTicketType={selectedTicketType}
        handleTicketTypeChange={handleTicketTypeChange}
        handleOrder={handleOrder}
        countdown={countdown}
        canOrder={canOrder}
        hasOrdered={hasOrdered}
        user={user}
      />
    );
  });

  test('Should render correctly', () => {
    expect(screen.getByTestId('event-detail')).toBeInTheDocument();
  });

  test('Should display correct location and date-time', () => {
    expect(screen.getByTestId('location-container')).toHaveTextContent(mockEvent.Location.namaProvinsi);
    expect(screen.getByTestId('date-time-container')).toHaveTextContent(formattedDate);
  });

  test('Should handle ticket quantity changes', () => {
    fireEvent.click(screen.getByText('-'));
    fireEvent.click(screen.getByText('+'));
    expect(handleTicketQuantityChange).toHaveBeenCalledTimes(2);
  });

  test('Should handle order button click', () => {
    fireEvent.click(screen.getByText('Beli Sekarang'));
    expect(handleOrder).toHaveBeenCalled();
  });

  test('Should match snapshot', () => {
    const dialog = screen.getByTestId('event-detail');
    expect(dialog).toMatchSnapshot();
  });
});
