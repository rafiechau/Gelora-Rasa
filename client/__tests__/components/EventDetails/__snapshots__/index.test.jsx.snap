// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetail Should match snapshot 1`] = `
<div
  class="eventDetail"
  data-testid="event-detail"
>
  <div
    class="containerLocation"
    data-testid="location-container"
  >
    <div
      class="containerPlace"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="PlaceIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
        />
      </svg>
      <span
        class="location"
      >
        Jakarta
      </span>
    </div>
    <div
      class="containerVenueName"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="LocationCityIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15 11V5l-3-3-3 3v2H3v14h18V11h-6zm-8 8H5v-2h2v2zm0-4H5v-2h2v2zm0-4H5V9h2v2zm6 8h-2v-2h2v2zm0-4h-2v-2h2v2zm0-4h-2V9h2v2zm0-4h-2V5h2v2zm6 12h-2v-2h2v2zm0-4h-2v-2h2v2z"
        />
      </svg>
      <span
        class="venueName"
      >
        Jalan Raya No.123, Jakarta
      </span>
    </div>
  </div>
  <div
    class="dateTime"
    data-testid="date-time-container"
  >
    <div
      class="containerDateTime"
    >
      <span
        class="date"
      >
        01 Jan 2024
      </span>
    </div>
    <div
      class="containerDateTime"
    >
      <span
        class="time"
      >
        20:00
      </span>
    </div>
    <div
      class="containerDateTime"
    >
      <span
        class="eventType"
      >
        hybrid
      </span>
    </div>
  </div>
  <div
    class="ticketInfo"
    data-testid="ticket-info-container"
  >
    <div
      class="containerTicketInfo"
    >
      <span
        class="ticketPrice"
      >
        RpÂ 50.000
      </span>
      <span
        class="ticketQuantity"
      >
        <button
          type="button"
        >
          -
        </button>
        <input
          type="number"
          value="1"
        />
        <button
          type="button"
        >
          +
        </button>
      </span>
    </div>
    <div
      class="ticketTypeSelector"
    >
      <label
        class="radioLabel"
        for="offline"
      >
        <input
          checked=""
          class="radioInput"
          id="offline"
          name="ticketType"
          type="radio"
          value="offline"
        />
        <span>
          app_detail_event_offline
        </span>
      </label>
      <label
        class="radioLabel"
        for="online"
      >
        <input
          class="radioInput"
          id="online"
          name="ticketType"
          type="radio"
          value="online"
        />
        <span>
          app_detail_event_online
        </span>
      </label>
    </div>
    <div
      class="containerButton"
    >
      <span>
        <span>
          app_detail_event_coutdown_description
        </span>
        5 days
         
      </span>
      <button
        class="buyNow"
        type="button"
      >
        Beli Sekarang
      </button>
    </div>
  </div>
</div>
`;
